ansible/roles/dashmate/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+  ansible.builtin.group:
+    name: '{{ dashmate_group }}'
+
+- name: Create dashd user
+  ansible.builtin.user:
+    name: '{{ dashmate_user }}'
+    comment: 'Dashmate user'
+    group: '{{ dashmate_group }}'
+    home: '{{ dashmate_home }}'
+    create_home: true
+    umask: '0002'
+
+- name: Create .dashcore directory for dashmate user
+  ansible.builtin.file:
+    path: '{{ dashmate_home }}/.dashcore'
+    state: directory
+    mode: 0755
+    owner: '{{ dashmate_user }}'
+    group: '{{ dashmate_group }}'
+
+- name: Write dash.conf file in /etc/
+  ansible.builtin.template:
+    src: dash.conf.j2
+    dest: /etc/dash.conf
+    owner: root
+    group: root
+    mode: 0644
+  register: dash_config_state
+
+- name: Symlink dash.conf to user dashmate directory to make rpc work
+  ansible.builtin.file:
+    path: '{{ dashmate_home }}/.dashcore/dash.conf'
+    src: '/etc/dash.conf'
+    state: link
+    force: true
+
+- name: Adding dashmate to group docker
+  user:
+    name: 'dashmate'
+    groups: docker
+    append: yes
+
+- name: Create all necessary config dirs
+  ansible.builtin.file:
+    path: '{{ item }}'
+    state: directory
+    owner: '{{ dashmate_user }}'
+    group: '{{ dashmate_group }}'
+    recurse: true
+    mode: 0750
+  loop:
+    - '{{ dashmate_config_dir }}/{{ dash_network_name }}'
+    - '{{ dashmate_config_dir }}/logs'
+    - '{{ dashmate_config_dir }}/ssl'
+    - '{{ dashmate_config_dir }}/logs/{{ dash_network_name }}'
+    - '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}'
+
+- name: Generate an OpenSSL private key
+  community.crypto.openssl_privatekey:
+    path: '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/private.key'
+    # dapi-envoy docker user
+    owner: 101
+    group: 101
+
+  # common name is required when loading into envoy
+- name: Generate an OpenSSL Certificate Signing Request
+  community.crypto.openssl_csr:
+    path:  '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/bundle.csr'
+    privatekey_path: '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/private.key'
+    common_name: '{{ inventory_hostname }}'
+
+- name: Generate a Self Signed OpenSSL certificate
+  community.crypto.x509_certificate:
+    path: '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/bundle.crt'
+    csr_path: '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/bundle.csr'
+    privatekey_path: '{{ dashmate_config_dir }}/ssl/{{ dash_network_name }}/private.key'
+    provider: selfsigned
+    # dapi-envoy docker user
+    owner: 101
+    group: 101
+
+- name: Write config.json file in ~/.dashmate
+  vars:
+    template_bootstrap_peers: "{{ groups.seed_nodes | reject('equalto', inventory_hostname) }}"
+  ansible.builtin.template:
+    src: dashmate.json.j2
+    dest: '{{ dashmate_config_dir }}/config.json'
+    owner: '{{ dashmate_user }}'
+    group: '{{ dashmate_group }}'
+    mode: 0644
+
+- name: Install some required dependencies for yarn build
+  ansible.builtin.package:
+    name: cmake
```

Comment:
Let's install `libzmq3-dev` here as well, otherwise `yarn install` will fail in some cases on Ubuntu on `aarch64`